### PR TITLE
Add support for Wind Barb rendering for mesh vector datasets

### PR DESCRIPTION
**Pull request by uclaros**
## Description
Wind Barbs are a common way to display wind strength and direction on maps.

This PR adds a new _Wind Barbs_ symbology type for mesh vector datasets.

image

Wind barbs encode the wind speed in knots (nautical miles per hour) so the user should select the units his data are in, or a custom multiplier that should be used to convert the units to knots.

### CODE REVIEW

**Review by PeterPetrik — COMMENTED** [review]
Implementation looks good to me, but it looks to me that the wings are somewhat too close to each other and it is difficult to see if it is triangle or 2 wings. Maybe we need more white space between them?

compare with for example this - it is more readable I think

**Comment by uclaros** [comment]
> Maybe we need more white space between them?

The distance between the barbs scales with the overall size. I've pasted a wind barb of similar size on that image and the distances look quite similar to me.
barbs

Since there is no strict specification for the styling of the wind barbs, the distance between lines and their angles could be modified for readability. However,  the attached hand-drawn ones was the only case I encountered where the shaft has a variable length depending on the number of pennants and one of the very few where the barbs angle (lines) does not match that of the pennant (triangle), which to me does not look pretty.

Maybe someone experienced in reading wind charts could chip in the discussion!

**Comment by PeterPetrik** [comment]
> Maybe someone experienced in reading wind charts could chip in the discussion!

That would be good :)

File: src/core/mesh/qgsmeshvectorrenderer.cpp [hunk 1]
```diff
@@ -518,10 +518,163 @@ QgsMeshVectorRenderer *QgsMeshVectorRenderer::makeVectorRenderer(
...
+  const double cosAlpha = cos( vectorAngle ) * mag( xVal );
+  const double sinAlpha = sin( vectorAngle ) * mag( xVal );
+
+  // Now determine the X and Y distances of the end of the line from the start
+  // Flip the Y axis (pixel vs real-world axis)
+  const double xDist = cosAlpha * shaftLength;
+  const double yDist = - sinAlpha * shaftLength;
+
+  if ( std::abs( xDist ) < 1 && std::abs( yDist ) < 1 )
+    return;
```
Comment (wonder-sk) [inline src/core/mesh/qgsmeshvectorrenderer.cpp:598]: is this "if" needed? if shaftLength has reasonable size, it should never return...?

**Review by wonder-sk — APPROVED** [review]

**Pull request merged**